src/app/atoms/check-box/che-checkbox.component.spec.ts

```diff
@@ -1,26 +1,311 @@
+            // Arrange
+            fixture = createComponent(CheCheckboxF1Component);
+            component = fixture.componentInstance;
+            fixture.detectChanges();
+        });
+
+
+        it('Should have a label attribute', () => {
+            // Arrange
+            let checkbox = fixture.debugElement.query(By.css('che-checkbox'))!;
+            // Act
+
+            // Assert
+            expect(checkbox.nativeElement.getAttribute('label')).toEqual('this is a label');
+
+        });
+
+
+    });
+
+
+    describe('If it is disabled', () => {
+
+        let component: any;
+        let fixture: ComponentFixture<any>;
+
+        beforeEach(() => {
+            // Arrange
+            fixture = createComponent(CheCheckboxF3Component);
+            component = fixture.componentInstance;
+            fixture.detectChanges();
+        });
+
+
+        it('Should have attribute disabled', () => {
+            // Arrange
+            let checkbox = fixture.debugElement.query(By.css('che-checkbox'))!;
+            // Act
+
+            // Assert
+            expect(checkbox.nativeElement.getAttribute('disabled')).toBeTruthy('Should have attribute attribute disabled');
+
+        });
+
+
+    });
+
+    describe('If it is unchecked and clicked', () => {
+
+        let component: any;
+        let fixture: ComponentFixture<any>;
+
+        beforeEach(() => {
+            // Arrange
+
+            fixture = createComponent(CheCheckboxF4Component);
+            component = fixture.debugElement.componentInstance;
+            fixture.detectChanges();
+
+        });
+
+
+        it('should change from unchecked to marked and transparent bg to bg color', () => {
+            // Arrange
+            let checkbox = fixture.debugElement.query(By.css('input'))!;
+            let style = window.getComputedStyle(checkbox.nativeElement, null);
+            // Assert
+
+            expect(style.getPropertyValue('background-color')).toEqual('rgba(0, 0, 0, 0)');
+            expect(checkbox.nativeElement.classList.contains('che-checkbox-checked')).toBeFalsy('Should have not class "che-checkbox-checked"');
+            // Act
+
+            component.isChecked = true;
+            fixture.detectChanges();
+            checkbox.nativeElement.dispatchEvent(new Event('change'));
+            fixture.detectChanges();
+            // Assert
+            expect(checkbox.nativeElement.classList.contains('che-checkbox-checked')).toBeTruthy('Should have class "che-checkbox-checked"');
+
+        });
+
+
+    });
+
+    describe('If it is checked and clicked', () => {
+
+        let component: any;
+        let fixture: ComponentFixture<any>;
+
+        beforeEach(() => {
+            // Arrange
+            fixture = createComponent(CheCheckboxF4Component);
+            component = fixture.componentInstance;
+            fixture.detectChanges();
+        });
+
+
+        it('should change from checked to unchecked and  bg color to transparent bg', () => {
```

Comment:
Should to change to unchecked and change from bg color to transparent bg to click.